app/client/src/pages/Editor/PropertyPane/PropertyControlsGenerator.tsx

```diff
@@ -29,11 +31,26 @@ export interface PropertyControlsGeneratorProps {
   searchQuery?: string;
 }
 
+export const shouldSectionBeExpanded = (
+  sectionConfig: PropertyPaneSectionConfig,
+  isFlagEnabled: boolean,
+) => {
+  if (isFlagEnabled && "expandedByDefault" in sectionConfig) {
+    return !!sectionConfig.expandedByDefault;
+  } else {
+    if (!!sectionConfig.isDefaultOpen) {
```

Comment:
Please correct me if I'm wrong: Seems like `if (!!sectionConfig.isDefaultOpen)` will always run if it is set to `true`?

Maybe we can do this:
```suggestion
    if (isFlagEnabled && sectionConfig.hasOwnProperty("expandedByDefault") {
      return sectionConfig.expandedByDefault;
    }
    return true;
```

Reply:
I see that @rishabhrathod01 has already mentioned it.

Reply:
Made the required changes